server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -514,12 +520,22 @@ public void releaseLock(long primaryTerm, long generation, String acquirerId) th
     @Override
     public Boolean isLockAcquired(long primaryTerm, long generation) throws IOException {
         String metadataFile = getMetadataFileForCommit(primaryTerm, generation);
-        return isLockAcquired(metadataFile);
+        return isLockAcquired(metadataFile, false);
     }
 
     // Visible for testing
-    Boolean isLockAcquired(String metadataFile) throws IOException {
-        return mdLockManager.isAcquired(FileLockInfo.getLockInfoBuilder().withFileToLock(metadataFile).build());
+    Boolean isLockAcquired(String metadataFile, boolean useCache) throws IOException {
+        if (useCache && acquiredLockCache.get(metadataFile) != null) {
+            return true;
+        }
+        boolean lockAcquired = mdLockManager.isAcquired(FileLockInfo.getLockInfoBuilder().withFileToLock(metadataFile).build());
+        if (lockAcquired) {
+            // We do not want to cache info of unlocked metadata files as we do not want to miss a lock getting acquired
+            // on a metadata file between 2 invocations of this method. Caching unlocked files can lead to data
+            // consistency issues.
```

Comment:
This is critical to our consistency guarantee. Can we please add a test that enforces this so that we catch any regressions introduced in the future? Right now, I can't think of a more aggressive mechanism to avoid such a regression.

Reply:
> Can we please add a test that enforces this so that we catch any regressions introduced in the future? 

But we are not caching absence of lock, right? Why does it cause the regression?